### PR TITLE
Update chain.schema.json

### PR DESCRIPTION
**Pull request by faddat**
bumps for chain.schema.json

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented) not applicable

### CODE REVIEW

File: chain.schema.json [hunk 1]
```diff
@@ -2,19 +2,19 @@
   "$schema": "http://json-schema.org/draft-07/schema#",
   "codebase":{
     "git_repo": "https://github.com/osmosis-labs/osmosis",
-    "recommended_version": "v12.2.0",
+    "recommended_version": "v12.3.0",
     "compatible_versions": [
-      "v12.2.0"
```
Comment (ValarDragon) [inline chain.schema.json:7]: Should include both, v12.2.0, v12.2.1, v12.3.0

File: chain.schema.json [hunk 2]
```diff
@@ -2,19 +2,19 @@
...
+      "v12.3.0"
     ],
     "binaries": {
-      "linux/amd64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.2.0/osmosisd-12.2.0-linux-amd64?checksum=sha256:b4a0296b142b1a535f3116021d39660868a83fc66b290ab0891b06211f86fd31",
-      "linux/arm64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.2.0/osmosisd-12.2.0-linux-arm64?checksum=sha256:84717e741b61ef3616fef403aae42067614e58a0208347cd642f6d03240b7778"
+      "linux/amd64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.2.0/osmosisd-12.3.0-linux-amd64?checksum=sha256:b4a0296b142b1a535f3116021d39660868a83fc66b290ab0891b06211f86fd31",
+      "linux/arm64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.2.0/osmosisd-12.3.0-linux-arm64?checksum=sha256:84717e741b61ef3616fef403aae42067614e58a0208347cd642f6d03240b7778"
```
Comment (ValarDragon) [inline chain.schema.json:11]: This is wrong, binary hashes are not updated.

File: chain.schema.json [hunk 3]
```diff
@@ -2,19 +2,21 @@
...
-      "linux/arm64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.2.0/osmosisd-12.2.0-linux-arm64?checksum=sha256:84717e741b61ef3616fef403aae42067614e58a0208347cd642f6d03240b7778"
+      "linux/amd64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.3.0/osmosisd-12.3.0-linux-amd64?checksum=sha256:958210c919d13c281896fa9773c323c5534f0fa46d74807154f737609a00db70",
+      "linux/arm64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.3.0/osmosisd-12.3.0-linux-arm64?checksum=sha256:a931618c8a839c30e5cecfd2a88055cda1d68cc68557fe3303fe14e2de3bef8f"
     },
     "cosmos_sdk_version": "0.45",
     "tendermint_version": "0.34",
     "cosmwasm_version": "0.28",
     "cosmwasm_enabled": true,
-    "ibc_go_version": "3.3.0",
+    "ibc_go_version": "3.3.1",
```
Comment (p0mvn) [inline chain.schema.json:19]: v12.3.0 seems to still be on 3.3.0:
https://github.com/osmosis-labs/osmosis/blob/7e41ec9e69bce66f519ae715d2a8adf74e377263/go.mod#L11

File: chain.schema.json [hunk 4]
```diff
@@ -2,19 +2,21 @@
...
-      "v12.2.0"
+      "v12.3.0",
+      "v12.2.0",
+      "v12.2.1"
     ],
     "binaries": {
-      "linux/amd64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.2.0/osmosisd-12.2.0-linux-amd64?checksum=sha256:b4a0296b142b1a535f3116021d39660868a83fc66b290ab0891b06211f86fd31",
-      "linux/arm64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.2.0/osmosisd-12.2.0-linux-arm64?checksum=sha256:84717e741b61ef3616fef403aae42067614e58a0208347cd642f6d03240b7778"
+      "linux/amd64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.3.0/osmosisd-12.3.0-linux-amd64?checksum=sha256:958210c919d13c281896fa9773c323c5534f0fa46d74807154f737609a00db70",
+      "linux/arm64": "https://github.com/osmosis-labs/osmosis/releases/download/v12.3.0/osmosisd-12.3.0-linux-arm64?checksum=sha256:a931618c8a839c30e5cecfd2a88055cda1d68cc68557fe3303fe14e2de3bef8f"
```
Comment (p0mvn) [inline chain.schema.json:13]: hashes lgtm

**Review by ValarDragon — APPROVED** [review]
TY!

(Though perhaps we should make a CI job to help us with these steps in the future)

**Pull request merged**